modules/contracts/apps/challenge/src/handlers/execute.rs

```diff
@@ -0,0 +1,497 @@
+        ChallengeExecuteMsg::CancelChallenge { challenge_id } => {
+            cancel_challenge(deps, info, &app, challenge_id)
+        }
+        ChallengeExecuteMsg::UpdateFriendsForChallenge {
+            challenge_id,
+            friends,
+            op_kind,
+        } => update_friends_for_challenge(deps, info, &app, challenge_id, friends, op_kind),
+        ChallengeExecuteMsg::DailyCheckIn {
+            challenge_id,
+            metadata,
+        } => daily_check_in(deps, env, info, &app, challenge_id, metadata),
+        ChallengeExecuteMsg::CastVote { vote, challenge_id } => {
+            cast_vote(deps, env, &app, vote, challenge_id)
+        }
+    }
+}
+
+/// Create new challenge
+fn create_challenge(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    app: ChallengeApp,
+    challenge_req: ChallengeRequest,
+) -> AppResult {
+    // Only the admin should be able to create a challenge.
+    app.admin.assert_admin(deps.as_ref(), &info.sender)?;
+
+    let mut challenge = ChallengeEntry::new(challenge_req);
+    challenge.set_total_check_ins(&env)?;
+
+    //check that the challenge status is ChallengeStatus::Uninitialized
+    if challenge.status != ChallengeStatus::Uninitialized {
+        return Err(AppError::WrongChallengeStatus {});
+    }
+
+    // Generate the challenge id and update the status
+    let challenge_id = NEXT_ID.update(deps.storage, |id| AppResult::Ok(id + 1))?;
+    challenge.status = ChallengeStatus::Active;
+    CHALLENGE_LIST.save(deps.storage, challenge_id, &challenge)?;
+
+    // Create the initial check_in entry
+    DAILY_CHECK_INS.save(
+        deps.storage,
+        challenge_id,
+        &vec![CheckIn::default_from(&env)],
+    )?;
+
+    // Create the initial challenge_votes entry
+    CHALLENGE_VOTES.save(deps.storage, challenge_id, &Vec::new())?;
+
+    Ok(app.tag_response(
+        Response::new().add_attribute("challenge_id", challenge_id.to_string()),
+        "create_challenge",
+    ))
+}
+
+fn update_challenge(
+    deps: DepsMut,
+    info: MessageInfo,
+    app: ChallengeApp,
+    challenge_id: u64,
+    new_challenge: ChallengeEntryUpdate,
+) -> AppResult {
+    app.admin.assert_admin(deps.as_ref(), &info.sender)?;
+
+    // will return an error if the challenge doesn't exist
+    let mut loaded_challenge: ChallengeEntry = CHALLENGE_LIST
+        .may_load(deps.storage, challenge_id)?
+        .ok_or(AppError::NotFound {})?;
+
+    if loaded_challenge.status != ChallengeStatus::Active {
+        return Err(AppError::WrongChallengeStatus {});
+    }
+
+    if let Some(name) = new_challenge.name {
+        loaded_challenge.name = name;
+    }
+    if let Some(collateral) = new_challenge.collateral {
+        loaded_challenge.collateral = collateral;
+    }
+    if let Some(description) = new_challenge.description {
+        loaded_challenge.description = description;
+    }
+    if let Some(end) = new_challenge.end {
+        loaded_challenge.end = end;
+    }
```

Comment:
Is it okay to update collateral and end for the challenge?

Reply:
Removed collateral and end as yes, they could be abused.